### DISCUSSION TITLE
FATAL: Required dependency "node" is using an unsupported version.

### DISCUSSION THREAD

**Discussion by Crazygiscool** [BlueprintFramework/framework#65] [Support] [answered]
when added all dependencies, and definitly node, also changed node version by "n latest", extracted everything from the release.zip, and replaced All. it gave me error when try to run sudo bash blueprint.sh, and yes, if you wanna ask, I also ran it without sudo too

FULL ERROR
  ██▌▌▌   Blueprint Framework
██  ██▌▌▌ https://blueprint.zip
  ████▌▌▌ © 2023-2024 Ivy (prpl.wtf)

20:36:49 INFO: Searching and validating framework dependencies..
20:36:49 FATAL: Some framework dependencies are not installed or detected.
20:36:49 FATAL: Required dependency "node" is using an unsupported version.

**Comment by heyitslumi** [comment]
could you show us the output of `node -v`?

**Reply by Crazygiscool** [reply]
nvm, I got it to work, I got the wrong version, but after all is installed it says 
You need to finish installing Blueprint to start using extensions.
in the extensions tab in the panel 
![Screen Shot 2024-11-18 at 4 48 54 PM](https://github.com/user-attachments/assets/d78d735c-beda-499a-83ef-d6a07b6de930)
above is a image

**Reply by Crazygiscool** [reply]
and when I did blueprint debug there was a error showing this
chown: cannot dereference '/var/www/pterodactyl/public/storage': No such file or directory
ty

**Reply by Crazygiscool** [reply]
so I updated the whole pterodactyl panel and wing, and blueprint extension button is now no where to be found, so I did sudo blueprint -rerun-install and it gave this
ERROR in resources/scripts/blueprint/extends/routers/ServerRouter.tsx:41:79
TS2339: Property 'BlueprintFramework' does not exist on type 'Server'.
    39 | export const NavigationLinks = () => {
    40 |   const rootAdmin = useStoreState((state) => state.user.data!.rootAdmin);
  > 41 |   const serverEgg = ServerContext.useStoreState((state) => state.server.data?.BlueprintFramework.eggId);
       |                                                                               ^^^^^^^^^^^^^^^^^^
    42 |   const match = useRouteMatch<{ id: string }>();
    43 |   const to = (value: string, url = false) => {
    44 |     if (value === '/') {

ERROR in resources/scripts/blueprint/extends/routers/ServerRouter.tsx:99:79
TS2339: Property 'BlueprintFramework' does not exist on type 'Server'.
     97 | export const NavigationRouter = () => {
     98 |   const rootAdmin = useStoreState((state) => state.user.data!.rootAdmin);
  >  99 |   const serverEgg = ServerContext.useStoreState((state) => state.server.data?.BlueprintFramework.eggId);
        |                                                                               ^^^^^^^^^^^^^^^^^^
    100 |   const match = useRouteMatch<{ id: string }>();
    101 |   const to = (value: string, url = false) => {
    102 |     if (value === '/') {
error Command failed with exit code 2.

**Reply by prplwtf** [reply]
Blueprint is not compatible with non-Blueprint modifications, which commonly causes that error.

**Comment by prplwtf** [comment, marked as answer]
Blueprint requires Node.js v17 or higher.
